packages/dds/sequence/src/sequence.ts

```diff
@@ -707,7 +706,11 @@ export abstract class SharedSegmentSequence<T extends ISegment>
                 // first we stop deferring incoming ops, and apply then all
                 this.deferIncomingOps = false;
                 while (this.loadedDeferredIncomingOps.length > 0) {
-                    this.processCore(this.loadedDeferredIncomingOps.shift(), false, undefined);
+                    const message = this.loadedDeferredIncomingOps.shift();
```

Comment:
this is pretty awfully inefficient--should do this instead:

```
for (const message of this.loadedDeferredIncomingOps) {
    this.processCore(message, false, undefined);
}
this.loadedDeferredIncomingOps.length = 0;
```

Reply:
(I know pre-existing, but having strict null checks makes the rough pattern worse; might as well clean it up while we're here)

Reply:
It's conditioned on `this.deferIncomingOps` anyway, which we set to `false` immediately above. It does delocalize some of the logic a bit more, but if we think that's a problem there are better ways to deal with it (ex: split off the set of deferred ops into a different array, assert after iteration that there aren't any newly deferred ones). I personally don't think it's an issue, the intent of things is clear enough.